include/parpeoptimization/minibatchOptimization.h

```diff
@@ -509,51 +509,94 @@ class MinibatchOptimizer {
                                                OptimizationReporter *reporter) {
         int maxSubsequentFails = 10;
         bool finalFail = false;
+        bool initialFail = false;
         cost = NAN;
         FunctionEvaluationStatus status = functionEvaluationFailure;
+        std::vector<double> parDifference;
```

Comment:
```suggestion
        std::vector<double> parDifference(parameters.size());
```
and remove .resize()